src/frontend/next/src/components/Posts/Timeline.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC } from 'react';
+
+import { Container, createStyles, Grid } from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import Post from '../Post/Post';
+import Spinner from '../Spinner/Spinner.jsx';
+import LoadAutoScroll from './LoadAutoScroll';
+import useSiteMetaData from '../../hooks/use-site-metadata';
+
+type Props = {
+  pages: Array<any> | undefined;
```

Comment:
This will be an Array of Post objects, so we should be able to type it with your interface.